app/controllers/FaciaToolController.scala

```diff
@@ -41,26 +41,41 @@ class FaciaToolController(
         Ok(Json.toJson(configJson)).as("application/json")}}}
 
   def getCollection(collectionId: String) = AccessAPIAuthAction.async { implicit request =>
-
     val collectionPriorities = configAgent.getFrontsPermissionsPriorityByCollectionId(collectionId)
 
     withModifyGroupPermissionForCollections(collectionPriorities, Set()) {
+      val collections = fetchCollections(List(collectionId))
 
-      FaciaToolMetrics.ApiUsageCount.increment()
-      val collection = frontsApi.amazonClient.collection(collectionId).flatMap { collectionJson =>
-        collectionJson.map(json => mediaServiceClient.addThumbnailsToCollection(json, collectionId)) match {
-          case Some(f) => f.map(Some(_))
-          case None => Future.successful(None)
-
-        }
-      }
-      collection.map(c => NoCache {
+      collections.head.map(c => NoCache {
         Ok(Json.toJson(c)).as("application/json")
       })
+    }
+  }
+
+  def getCollections(collectionIdsString: String) = AccessAPIAuthAction.async { implicit request =>
+    val collectionIds = collectionIdsString.split(",").toList
+    val collectionPriorities = collectionIds.flatMap(configAgent.getFrontsPermissionsPriorityByCollectionId(_)).toSet
+
+    withModifyGroupPermissionForCollections(collectionPriorities, Set()) {
+      val collections = fetchCollections(collectionIds)
 
+      Future.sequence(collections).map(c => NoCache {
+        Ok(Json.toJson(c)).as("application/json")
+      })
     }
   }
 
+  def fetchCollections(collectionIds: List[String]) = {
+    FaciaToolMetrics.ApiUsageCount.increment()
+    collectionIds.map(collectionId => frontsApi.amazonClient.collection(collectionId).flatMap { collectionJson =>
```

Comment:
`amazonClient` here is still making multiple requests. Can this make a bulk request? This may give you a similar optimisation as you intended with the call to your own server.

Reply:
I'd love toI did some searching before opening the PR and every answer to the question 'can S3 handle requests for multiple files' was met with a negative. Admittedly most of the replies are old. I've asked around and can't find an answer to the contrary, but if there was a way I'd be delighted to use it!